framework/src/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -522,36 +697,38 @@ private AnnotationMirror createNumberAnnotationMirror(List<Number> values) {
-        List<? extends Number> lefts = getNumericalValues(leftNode, p);
-        List<? extends Number> rights = getNumericalValues(rightNode, p);
-        if (lefts == null || rights == null) {
-            return null;
-        }
         List<Boolean> resultValues = new ArrayList<>();
-        for (Number left : lefts) {
-            NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
-            for (Number right : rights) {
-                switch (op) {
-                    case EQUAL:
-                        resultValues.add(nmLeft.equalTo(right));
-                        break;
-                    case GREATER_THAN:
-                        resultValues.add(nmLeft.greaterThan(right));
-                        break;
-                    case GREATER_THAN_EQ:
-                        resultValues.add(nmLeft.greaterThanEq(right));
-                        break;
-                    case LESS_THAN:
-                        resultValues.add(nmLeft.lessThan(right));
-                        break;
-                    case LESS_THAN_EQ:
-                        resultValues.add(nmLeft.lessThanEq(right));
-                        break;
-                    case NOT_EQUAL:
-                        resultValues.add(nmLeft.notEqualTo(right));
-                        break;
-                    default:
-                        throw new UnsupportedOperationException();
+        if (!isIntRange(leftNode, p) && !isIntRange(rightNode, p)) {
+            List<? extends Number> lefts = getNumericalValues(leftNode, p);
+            List<? extends Number> rights = getNumericalValues(rightNode, p);
+            if (lefts == null || rights == null) {
+                return null;
+            }
+            for (Number left : lefts) {
+                NumberMath<?> nmLeft = NumberMath.getNumberMath(left);
+                for (Number right : rights) {
+                    switch (op) {
+                        case EQUAL:
+                            resultValues.add(nmLeft.equalTo(right));
+                            break;
+                        case GREATER_THAN:
+                            resultValues.add(nmLeft.greaterThan(right));
+                            break;
+                        case GREATER_THAN_EQ:
+                            resultValues.add(nmLeft.greaterThanEq(right));
+                            break;
+                        case LESS_THAN:
+                            resultValues.add(nmLeft.lessThan(right));
+                            break;
+                        case LESS_THAN_EQ:
+                            resultValues.add(nmLeft.lessThanEq(right));
+                            break;
+                        case NOT_EQUAL:
+                            resultValues.add(nmLeft.notEqualTo(right));
+                            break;
+                        default:
+                            throw new UnsupportedOperationException();
+                    }
                 }
```

Comment:
Oh I was taking about typetools/master#1162. Should I put this info into the comment?